pkg/resources/reconciler_test.go

```diff
@@ -559,6 +563,30 @@ func TestReconciler_ReconcileNamespace(t *testing.T) {
 			Installation:   &integreatlyv1alpha1.RHMI{},
 			ExpectedStatus: integreatlyv1alpha1.PhaseInProgress,
 		},
+		{
+			Name: "Test namespace reconcile return error when pulling secret fails",
+			client: fakeclient.NewFakeClientWithScheme(scheme, &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: nsName,
+					Labels: map[string]string{
+						OwnerLabelKey: string(installation.GetUID()),
+					},
+				},
+				Status: corev1.NamespaceStatus{
+					Phase: corev1.NamespaceActive,
+				},
+			}),
+			Installation: &integreatlyv1alpha1.RHMI{
+				Spec: integreatlyv1alpha1.RHMISpec{
+					PullSecret: integreatlyv1alpha1.PullSecretSpec{
+						Name:      "test",
+						Namespace: "test",
+					},
+				},
+			},
+			ExpectedStatus: integreatlyv1alpha1.PhaseFailed,
+			ExpectErr:      true,
+		},
```

Comment:
hey @bryankeane0 
I am a bit confused, wonder if I'm missing something.
This unit test seems to cover a logic that's 2 years old. That's not bad but I feel like we could test here the following scenarios:
1. Is the label added to a ns that doesn't have it
2. Is the label updated if an ns has it set to true